spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ActuatorHealthApplicationStatusProvider.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.consul.serviceregistry;
+
+import java.util.Optional;
+
+import com.ecwid.consul.v1.health.model.Check.CheckStatus;
+
+import org.springframework.boot.actuate.health.HealthComponent;
+import org.springframework.boot.actuate.health.HealthEndpoint;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.cloud.consul.discovery.HeartbeatProperties;
+
+import static org.springframework.boot.actuate.health.Status.DOWN;
+import static org.springframework.boot.actuate.health.Status.OUT_OF_SERVICE;
+import static org.springframework.boot.actuate.health.Status.UP;
+
+/**
+ * Leverages Spring Boot Actuator health endpoint to determine the current health of the
+ * application.
+ *
+ * @author Chris Bono
+ */
+public class ActuatorHealthApplicationStatusProvider implements ApplicationStatusProvider {
```

Comment:
I originally created an additional Reactive status provider implementation that used the Reactive contributor registry to determine the application status. I switched to using the `HealthEndpoint` for both reactive and imperative modes because of the following:

1. The HealthEndpoint will adapt all reactive indicators into an imperative adapted version when [Flux is on the classpath](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/health/HealthEndpointConfiguration.java#L83).

2. When dealing w/ a reactive and non-reactive world, at some point the membrane is reached between the two worlds. The further that boundary is, the bigger the area of the code that is Reactive, the more benefit we gain from Reactive. At some point the Mono/Flux will be subscribed and resolved into an actual value(s). The TTL scheduler just runs a task in its own separate thread every N seconds that will ask the health system for its status. Because of the non-reactive nature of the scheduler there is little value in hooking into the Reactive registry as the non-reactive adapts the indicators. This reduces the complexity of the configuration and maintenance for us.

If the TTLScheduler is modified to have a Reactive and non-reactive version then it would make sense to create a separate `ReactiveApplicationStatusProvider` that uses the `ReactiveHealthEndpointWebExtension` for its information.